### PR TITLE
[1.16.5] Sign Refactor

### PR DESCRIPTION
**Pull request by JacksonPlayz**
This PR refactors signs to use the vanilla `WoodType` which allows for the use of the default vanilla sign classes (better compatibility). Custom standing and wall block classes along with a custom tile entity class is still required to use our tile entity so we can register new sign blocks.

I've also backported a change from my previous sign refactor PR which makes `TileEntitySubRegistryHelper#collectBlocks` take in any class instead of one that extends block as `AbnormalsAbstractSignBlock` no longer exists.

### CODE REVIEW

File: src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java [hunk 1]
```diff
@@ -72,6 +77,7 @@
 	public static final EndimationDataManager ENDIMATION_DATA_MANAGER = new EndimationDataManager();
 	public static final RegistryHelper REGISTRY_HELPER = new RegistryHelper(MODID);
 	public static final TrackedData<Byte> SLABFISH_SETTINGS = TrackedData.Builder.create(DataProcessors.BYTE, () -> (byte) 8).enablePersistence().build();
+	public static final Set<WoodType> WOOD_TYPES = new HashSet<>();
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java:80]: Not a fan of this field being in `AbnormalsCore`.
A `SignManager` class would be better.

File: src/main/java/com/minecraftabnormals/abnormals_core/core/util/registry/BlockSubRegistryHelper.java [hunk 2]
```diff
@@ -213,17 +213,18 @@ public BlockSubRegistryHelper(RegistryHelper parent, DeferredRegister<Item> item
...
 	 */
 	public Pair<RegistryObject<AbnormalsStandingSignBlock>, RegistryObject<AbnormalsWallSignBlock>> createSignBlock(String name, MaterialColor color) {
-		ResourceLocation texture = new ResourceLocation(this.parent.getModId(), "textures/entity/signs/" + name + ".png");
-		RegistryObject<AbnormalsStandingSignBlock> standing = this.deferredRegister.register(name + "_sign", () -> new AbnormalsStandingSignBlock(Block.Properties.create(Material.WOOD).doesNotBlockMovement().hardnessAndResistance(1.0F).sound(SoundType.WOOD), texture));
-		RegistryObject<AbnormalsWallSignBlock> wall = this.deferredRegister.register(name + "_wall_sign", () -> new AbnormalsWallSignBlock(Block.Properties.create(Material.WOOD, color).doesNotBlockMovement().hardnessAndResistance(1.0F).sound(SoundType.WOOD).lootFrom(standing.get()), texture));
-		this.itemRegister.register(name + "_sign", () -> new AbnormalsSignItem(standing.get(), wall.get(), new Item.Properties().maxStackSize(16).group(ItemGroup.DECORATIONS)));
+		WoodType type = WoodType.register(WoodType.create(this.parent.getModId() + ":" + name));
+		AbnormalsCore.WOOD_TYPES.add(type);
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/util/registry/BlockSubRegistryHelper.java:224]: This is not safe during parallel mod loading.
The `HashSet` could become corrupted easily. This is more reason to add a `SignManager` class with threadsafe methods.

**Review by SmellyModder — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecraftabnormals/abnormals_core/core/api/SignManager.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import net.minecraft.block.WoodType;
+import net.minecraft.client.renderer.Atlases;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SignManager {
+    public static final Set<WoodType> WOOD_TYPES = new HashSet<>();
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/api/SignManager.java:12]:
```suggestion
    private static final Set<WoodType> WOOD_TYPES = new HashSet<>();
```

File: src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java [hunk 4]
```diff
@@ -72,6 +78,7 @@
...
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java:81]: This field is still here.

File: src/main/java/com/minecraftabnormals/abnormals_core/core/api/SignManager.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import net.minecraft.block.WoodType;
+import net.minecraft.client.renderer.Atlases;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SignManager {
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/api/SignManager.java:11]:
```suggestion
public final class SignManager {
```

File: src/main/java/com/minecraftabnormals/abnormals_core/core/util/registry/BlockSubRegistryHelper.java [hunk 6]
```diff
@@ -5,21 +5,22 @@
 import com.minecraftabnormals.abnormals_core.common.blocks.chest.AbnormalsTrappedChestBlock;
 import com.minecraftabnormals.abnormals_core.common.blocks.sign.AbnormalsStandingSignBlock;
 import com.minecraftabnormals.abnormals_core.common.blocks.sign.AbnormalsWallSignBlock;
-import com.minecraftabnormals.abnormals_core.common.items.AbnormalsSignItem;
 import com.minecraftabnormals.abnormals_core.common.items.FuelBlockItem;
 import com.minecraftabnormals.abnormals_core.common.items.InjectedBlockItem;
 import com.minecraftabnormals.abnormals_core.common.tileentity.AbnormalsChestTileEntity;
 import com.minecraftabnormals.abnormals_core.common.tileentity.AbnormalsTrappedChestTileEntity;
+import com.minecraftabnormals.abnormals_core.core.AbnormalsCore;
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/util/registry/BlockSubRegistryHelper.java:12]: Unused import.

**Review by SmellyModder — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java [hunk 7]
```diff
@@ -33,7 +34,10 @@
 
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockState;
+import net.minecraft.block.WoodType;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.Atlases;
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java:39]: Unused import.

File: src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java [hunk 8]
```diff
@@ -33,7 +34,10 @@
 
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockState;
+import net.minecraft.block.WoodType;
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java:37]: Unused import.

**Review by SmellyModder — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java (hunk 7 above)
Comment (JacksonPlayz) [reply src/main/java/com/minecraftabnormals/abnormals_core/core/AbnormalsCore.java:39]: trollge

**Review by SmellyModder — APPROVED** [review]

**Pull request merged**